Stats and resources

AARandom opened the pull request:
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Included resources from gdrive (exam guide, notes, modules, trial papers, textbook as per resource team's request)
- Included thumbnails from spider.py
- Added subsection support for resources page and file
- Turned subsections in page into dropdown menu for easier scrolling on smaller screens

Fixes #3

## Type of change

- Content update
- New feature (non-breaking change which adds functionality)

*Please delete options that are not relevant.*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested the website locally with my changes, and I am sure that it works well on different screen sizes

AARandom commented:
Can the page load that many png? If cant then just include a few will do, we still provide links down there to google Drive. Idk, I didnt try...

Can push directly to master bah, for those images, cauz my browser crased alr 🤣 . Code baru make a PR

AARandom commented:
It worked on my local copy (typical bad omen saying, ik) but let me deploy a githup page for yall to check

AARandom commented:
> Jared: Can the page load that many png? If cant then just include a few will do, we still provide links down there to google Drive. Idk, I didnt try...
> 
> Can push directly to master bah, for those images, cauz my browser crased alr 🤣 . Code baru make a PR

Tbh my browser crashed a few times too just to create the PR. Avoid the compare tab at all costs 💀💀

AARandom commented:
@raihanahhusna can ya check pls tqq

jaredliw commented:
> > Jared: Can the page load that many png? If cant then just include a few will do, we still provide links down there to google Drive. Idk, I didnt try...
> > Can push directly to master bah, for those images, cauz my browser crased alr 🤣 . Code baru make a PR
> 
> Tbh my browser crashed a few times too just to create the PR. Avoid the compare tab at all costs 💀💀

I cherry-pick that large commit to `master`, u try to update your branch from `master`, see if it works?

AARandom commented:
> > > Jared: Can the page load that many png? If cant then just include a few will do, we still provide links down there to google Drive. Idk, I didnt try...
> > > Can push directly to master bah, for those images, cauz my browser crased alr 🤣 . Code baru make a PR
> > 
> > 
> > Tbh my browser crashed a few times too just to create the PR. Avoid the compare tab at all costs 💀💀
> 
> I cherry-pick that large commit to `master`, u try to update your branch from `master`, see if it works?

I dont think you merged the resources.html file (need that for dropdowns) tho.

AARandom commented:
But on a side note, I removed about 600-800 links mostly based on most likely file to get copyright claimed, as well as cherry picking some states to be removed.
The images load quicker now (on my end, at least)

Imma run the spider overnight and commit it to this branch tomorrow morning

jaredliw commented:
> But on a side note, I removed about 600-800 links mostly based on most likely file to get copyright claimed, as well as cherry picking some states to be removed.
> The images load quicker now (on my end, at least)
> 
> Imma run the spider overnight and commit it to this branch tomorrow morning
> 

wait, what copyright?

Did the cherry pick work? Or maybe u should do it urself bah, to the master branch.

Wont u get banned by google if running overnight?

AARandom commented:
> > But on a side note, I removed about 600-800 links mostly based on most likely file to get copyright claimed, as well as cherry picking some states to be removed.
> > The images load quicker now (on my end, at least)
> > Imma run the spider overnight and commit it to this branch tomorrow morning
> 
> wait, what copyright?
> 
> Did the cherry pick work? Or maybe u should do it urself bah, to the master branch.
> 
> Wont u get banned by google if running overnight?

I scared later copyright since some of the modules are straight up vector pdfs of books from Pelangi, Sasbadi and Nilam. Some of them got the answers in blue also which only the teacher should have access to. I think its best to let the students find it themselves in the google drive rather than publicizing it on the website.

The cherry picking worked, just that you didnt pick the resources.html file which I changed to add support for categories

And I dont think i'll get banned from google? idk. But i added a delay between requests  just in case

jaredliw commented on layouts/page/resources.html:
```diff
@@ -25,41 +25,98 @@ <h2 class="text-center text-primary fw-lighter">
+                                    <!-- Each category in a subject is a dropdown menu (accordion component) by it self -->
+                                    <!-- Each dropdown ust have unique id, even in different tabs -->
+                                    <div class="accordion">
+                                        <div class="accordion-item" id={{ $accordianCounter }}>
```
```suggestion
                                        <div class="accordion-item" id={{ $accordionCounter }}>
```

jaredliw commented on layouts/page/resources.html:
```diff
@@ -25,41 +25,98 @@ <h2 class="text-center text-primary fw-lighter">
         <div class="tab-content">
             {{ $fileLink := "https://drive.google.com/file/d/%s/view?usp=drive_link" }}
             {{ $firstTab := true }}
+            {{ $accordianCounter := 1 }}
```
```suggestion
            {{ $accordionCounter := 1 }}
```

jaredliw commented on layouts/page/resources.html:
```diff
@@ -25,41 +25,98 @@ <h2 class="text-center text-primary fw-lighter">
+                                    <!-- Each dropdown ust have unique id, even in different tabs -->
+                                    <div class="accordion">
+                                        <div class="accordion-item" id={{ $accordianCounter }}>
+                                            <h2 class="accordion-header" id={{ print "heading-" $accordianCounter }}>
```
```suggestion
                                            <h2 class="accordion-header" id={{ print "heading-" $accordionCounter }}>
```

jaredliw commented on layouts/page/resources.html:
```diff
@@ -25,41 +25,98 @@ <h2 class="text-center text-primary fw-lighter">
+                                    <div class="accordion">
+                                        <div class="accordion-item" id={{ $accordianCounter }}>
+                                            <h2 class="accordion-header" id={{ print "heading-" $accordianCounter }}>
+                                                <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target={{ print "#body-" $accordianCounter }} aria-expanded="true" aria-controls={{ print "body-" $accordianCounter }}>
```
```suggestion
                                                <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target={{ print "#body-" $accordionCounter }} aria-expanded="true" aria-controls={{ print "body-" $accordionCounter }}>
```

Expected: Collapsed btn has gray text, open then green
Acutal: A btn that has not clicked before is green, open still green, closed then turn back gray

If not mistaken adding `collapsed` will do lah (by inspecting the html), u check "join us" page accordion

One more thing is the border, kinda wierd frankly speaking (which is the default behavior of bs), u check "join us" page as well. I forgot how did i change...

jaredliw commented on layouts/page/resources.html:
```diff
@@ -25,41 +25,98 @@ <h2 class="text-center text-primary fw-lighter">
+                                                    {{ .subsection_name }}
+                                                </button>
+                                            </h2>
+                                            <div id={{ print "body-" $accordianCounter }} class="accordion-collapse collapse" aria-labelledby={{ print "heading-" $accordianCounter }} data-bs-parent={{ print "#" $accordianCounter }}>
```
```suggestion
                                            <div id={{ print "body-" $accordionCounter }} class="accordion-collapse collapse" aria-labelledby={{ print "heading-" $accordionCounter }} data-bs-parent={{ print "#" $accordionCounter }}>
```

jaredliw commented on layouts/page/resources.html:
```diff
@@ -25,41 +25,98 @@ <h2 class="text-center text-primary fw-lighter">
-                                </div>
-                            {{ end }}
-                        </div>
+                                    {{ $accordianCounter = add $accordianCounter 1 }}
```
```suggestion
                                    {{ $accordionCounter = add $accordionCounter 1 }}
```

jaredliw commented on layouts/page/resources.html:
```diff
@@ -25,41 +25,98 @@ <h2 class="text-center text-primary fw-lighter">
+                                            </h2>
+                                            <div id={{ print "body-" $accordianCounter }} class="accordion-collapse collapse" aria-labelledby={{ print "heading-" $accordianCounter }} data-bs-parent={{ print "#" $accordianCounter }}>
+                                                <div class="accordion-body">
+                                                    <!-- Gallery component inside the accordian body -->
```
```suggestion
                                                    <!-- Gallery component inside the accordion body -->
```

jaredliw commented on layouts/page/resources.html:
```diff
@@ -25,41 +25,98 @@ <h2 class="text-center text-primary fw-lighter">
+                        {{ if .has_subsection }}
+                            <div class="d-grid pb-2">
+                                {{ range .subsection }}
+                                    <!-- Each category in a subject is a dropdown menu (accordion component) by it self -->
```
```suggestion
                                    <!-- Each category in a subject is a dropdown menu (accordion component) by itself -->
```

jaredliw commented on layouts/page/resources.html:
```diff
@@ -25,41 +25,98 @@ <h2 class="text-center text-primary fw-lighter">
+                            <div class="d-grid pb-2">
+                                {{ range .subsection }}
+                                    <!-- Each category in a subject is a dropdown menu (accordion component) by it self -->
+                                    <!-- Each dropdown ust have unique id, even in different tabs -->
```
```suggestion
                                    <!-- Each dropdown must have unique id, even in different tabs -->
```

jaredliw reviewed (changes requested):
Marvellous!!!

I think this improvement really make the page nicer and less bulky. Thanks a lot!

Nothing much, I just change a spelling, and a suggestion on the border thingy

jaredliw commented:
> > > But on a side note, I removed about 600-800 links mostly based on most likely file to get copyright claimed, as well as cherry picking some states to be removed.
> > > The images load quicker now (on my end, at least)
> > > Imma run the spider overnight and commit it to this branch tomorrow morning
> > 
> > 
> > wait, what copyright?
> > Did the cherry pick work? Or maybe u should do it urself bah, to the master branch.
> > Wont u get banned by google if running overnight?
> 
> I scared later copyright since some of the modules are straight up vector pdfs of books from Pelangi, Sasbadi and Nilam. Some of them got the answers in blue also which only the teacher should have access to. I think its best to let the students find it themselves in the google drive rather than publicizing it on the website.
> 
> The cherry picking worked, just that you didnt pick the resources.html file which I changed to add support for categories
> 
> And I dont think i'll get banned from google? idk. But i added a delay between requests just in case

Shall we tell the resource dept? Idk this.

Ok, luckily it worked. just leave the others there will do, as long as we can focus on the code side.

That's good. Did u face any files that have no thumbnails? I suppose Google banned me (for a short period) cauz i downloaded large files to generate thumbnails locally, else it should be ok.

AARandom commented:
Thanks for the review Jared! That accordion spelling flew over my head omg 💀💀

AARandom commented:
> > > > But on a side note, I removed about 600-800 links mostly based on most likely file to get copyright claimed, as well as cherry picking some states to be removed.
> > > > The images load quicker now (on my end, at least)
> > > > Imma run the spider overnight and commit it to this branch tomorrow morning
> > > 
> > > 
> > > wait, what copyright?
> > > Did the cherry pick work? Or maybe u should do it urself bah, to the master branch.
> > > Wont u get banned by google if running overnight?
> > 
> > 
> > I scared later copyright since some of the modules are straight up vector pdfs of books from Pelangi, Sasbadi and Nilam. Some of them got the answers in blue also which only the teacher should have access to. I think its best to let the students find it themselves in the google drive rather than publicizing it on the website.
> > The cherry picking worked, just that you didnt pick the resources.html file which I changed to add support for categories
> > And I dont think i'll get banned from google? idk. But i added a delay between requests just in case
> 
> Shall we tell the resource dept? Idk this.
> 
> Ok, luckily it worked. just leave the others there will do, as long as we can focus on the code side.
> 
> That's good. Did u face any files that have no thumbnails? I suppose Google banned me (for a short period) cauz i downloaded large files to generate thumbnails locally, else it should be ok.

I think its okay since they probably sourced it from telegram study groups. Plus, those should be from 2021-is, so the publishers probably updated it since then.

But never thought google would ban tho, omg now im scared a bit

As for thumbnails, every file pdf file has them but i added an mp3 just in case and it came out like this (but still can access from the website tho)
![image](https://github.com/user-attachments/assets/1e57ab0f-73fa-4e1f-8ca3-4e21ab5bb87d)

AARandom commented:
Okay, done fixing the yellow background thing. Apparently its just add .accordion-flush

jaredliw reviewed (approved):
lgtm
